### PR TITLE
Help the user to figure out which conversion raises an error

### PR DESCRIPTION
**Pull request by stephane**
Display the key of the associated value on error.

Not sure about the proper way to handle that, logger for output or better check of value before cast but you get the idea...

### CODE REVIEW

File: getconf/base.py [hunk 1]
```diff
@@ -258,7 +258,11 @@ def getint(self, key, default=0, doc=''):
         value = self._get(key, default=default, doc=doc, type_hint='int')
         if value is None:
             return None
-        return int(value)
+        try:
+            return int(value)
+        except ValueError:
+            raise ValueError(
+                "Unable to cast '%s' as integer for the key %s." % (value, key))
```
Comment (dbaty) [inline getconf/base.py]: We could use `repr` and remove the quotes around "%s". Numeric values would not be displayed the same as  strings, which would be a bit clearer:

    "Unable to cast %s as integer for the key %s." % (repr(value), key))

Comment (xavfernandez) [inline getconf/base.py]: It's a shame to lose the traceback...
We could either use something like `six.reraise` or simply `logger.error` with the `key` and `value`.

**Comment by stephane** [comment]
@xavfernandez `six` is not a dependency of the project so I used logger.exception and raise, what do you think about that?

**Comment by xavfernandez** [comment]
@stephane we could take the interesting bit from six and put it in getconf.compat ?
But your version seems fine also, maybe simply use `raise` to reraise the same exception instead of `raise ValueError` ?

**Comment by stephane** [comment]
Not compatible with Python 2.6....

**Comment by fbochu** [comment]
``` Python
try:
    raise ValueError('very important message')
except ValueError as e:
    logging.exception(e)
    raise e
```
Does not work in Python 2.6 ?

**Comment by RemiCardona** [comment]
```python
except ValueError as e:
    raise e
```
Please don't ever ever ever do that. `raise` or `raise SomeOtherException`.

Also, how about ditching 2.6 completely? The issue here is the test, not the `raise` statement. `assertRaises()` can be used as a context manager only with 2.7+.

**Comment by stephane** [comment]
@RemiCardona, you're right: raise e -> raise, done.

In waiting for Python 2.6 ditching, I skipped tests with assertRaises() for versions of Python less than 2.7...
What do you think about that?

File: tests/test_base.py [hunk 2]
```diff
@@ -194,11 +194,20 @@ def test_environ_overrides_config(self):
     def test_get_defaults_dict(self):
         """Test fetching from the default config dict with non-string values."""
         warnings.warn("Use of get() directly is deprecated. Use .getstr() instead", DeprecationWarning)
-        getter = getconf.ConfigGetter('TESTNS', defaults={'DEFAULT':{"test":'54'}})
+        getter = getconf.ConfigGetter('TESTNS', defaults={'DEFAULT': {"test": '54', 'wrong': 'bad'}})
         self.assertEqual('54', getter.get('test'))
         self.assertEqual('54', getter.getstr('test'))
         self.assertEqual(54, getter.getint('test'))
+
+        if sys.version_info > (2, 6):
```
Comment (fbochu) [inline tests/test_base.py]:
``` Python
self.assertRaise(ValueError, getter.getint, 'wrong')
```

**Comment by stephane** [comment]
If everyone is agree to keep Python 2.6 support, I'll merge the version with ugly tests (after rebase).

File: tests/test_base.py [hunk 3]
```diff
@@ -194,11 +194,18 @@ def test_environ_overrides_config(self):
...
         """Test fetching from the default config dict with non-string values."""
         warnings.warn("Use of get() directly is deprecated. Use .getstr() instead", DeprecationWarning)
-        getter = getconf.ConfigGetter('TESTNS', defaults={'DEFAULT':{"test":'54'}})
+        getter = getconf.ConfigGetter('TESTNS', defaults={'DEFAULT': {"test": '54', 'wrong': 'bad'}})
         self.assertEqual('54', getter.get('test'))
         self.assertEqual('54', getter.getstr('test'))
         self.assertEqual(54, getter.getint('test'))
+
+        # Ugly test compatible with Python 2.6...
+        self.assertRaise(ValueError, getter.getint, 'wrong')
```
Comment (RemiCardona) [inline tests/test_base.py]: manque un `s`

**Comment by rbarrois** [comment]
I don't mind dropping support for Python2.6 in a future release.

In that case, what about the following path:
* Formally document that version 1.5 is the last release supporting Python2.6
* Versions from 1.6 onwards only support Python2.7+

And, why not:
* Support for Python2.7 will be dropped by the end of 2017

**Comment by RemiCardona** [comment]
As for the original try/except/raise bit, if we don't want to use `six`, I say we go with @stephane's original patch. With python3 we'll get implicit exception chaining and python2 users can figure it out themselves.

**Comment by stephane** [comment]
OK so I'm going to remove travis CI for 2.6, document getconf 1.6+ only support Python 2.7+ and merge my original patch (first of 3 of this PR).

**Comment by stephane** [comment]
Could a maintainer upload v1.5.1 on pypi, please?

**Comment by vperron** [comment]
Uploaded on pypi.

**Comment by stephane** [comment]
Thank you

**Pull request closed without merging**